bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/rest/JettyAdminService.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pihole.internal.rest;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.openhab.binding.pihole.internal.rest.model.DnsStatistics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * @author Martin Grzeslowski - Initial contribution
+ */
+@NonNullByDefault
+public class JettyAdminService implements AdminService {
+    private static final Gson GSON = new GsonBuilder()
+            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    private final Logger logger = LoggerFactory.getLogger(JettyAdminService.class);
+    private final String token;
+    private final URI baseUrl;
+    private final HttpClient client;
+
+    public JettyAdminService(String token, URI baseUrl, HttpClient client) {
+        this.token = token;
+        this.baseUrl = baseUrl;
+        this.client = client;
+        if (this.client.isStopped()) {
+            throw new IllegalStateException("HttpClient is stopped");
+        }
+    }
+
+    @Override
+    public Optional<DnsStatistics> summary() throws ExecutionException, InterruptedException, TimeoutException {
+        logger.debug("Getting summary");
+        var url = baseUrl.resolve("/admin/api.php?summaryRaw&auth=" + token);
+        var request = client.newRequest(url);
+        var response = request.send();
```

Comment:
Can you add an explicit timeout to the request?

Applies to all requests in this class.